### DISCUSSION TITLE
How to transform object schema's output?

### DISCUSSION THREAD

**Discussion by h-sigma** [react-hook-form/react-hook-form#10012] [Q&A]
```ts
export const jsonResource = <T extends ZodTypeAny>(schema: T) => {
    return z
        .object({
            data: schema,
        })
        .transform(val => val.data);
};
```
Compiler refuses to pick up `val.data` as a possible data. Is this possible with/without extreme type-acrobatics? All I want to do is unwrap the json data by a key.

**Comment by leapful** [comment]
@h-sigma 

If your input schema is `ZodTypeAny`, just put directly ZodTypeAny so `transform` can infer correctly the type definition.

Example: https://codesandbox.io/s/clever-rain-le3cxd?file=/src/App.tsx

**Reply by h-sigma** [reply]
However, then I'd lose the type inference on the returned schema beyond the `data` property.

**Comment by h-sigma** [comment]
Not proud of it:
```ts
export const jsonResource = <T extends ZodTypeAny>(schema: T) => {
    return z
        .object({
            data: schema,
        })
        .transform(val => (val as { data: T['_output'] }).data);
};

```
